lib/matplotlib/patches.py

```diff
@@ -1659,14 +1661,34 @@ def theta_stretch(theta, scale):
             theta = np.deg2rad(theta)
             x = np.cos(theta)
             y = np.sin(theta)
-            return np.rad2deg(np.arctan2(scale * y, x))
-        theta1 = theta_stretch(self.theta1, width / height)
-        theta2 = theta_stretch(self.theta2, width / height)
+            stheta = np.rad2deg(np.arctan2(scale * y, x))
+            # arctan2 has the range [-pi, pi], we expect [0, 2*pi]
+            return (stheta + 360) % 360
+
+        theta1 = self.theta1
+        theta2 = self.theta2
+
+        if (
+            # if we need to stretch the angles because we are distorted
+            width != height
+            # and we are not doing a full circle
+            and not (theta1 != theta2 and theta1 % 360 == theta2 % 360)
+        ):
+            theta1 = theta_stretch(self.theta1, width / height)
+            theta2 = theta_stretch(self.theta2, width / height)
 
         # Get width and height in pixels
-        width, height = self.get_transform().transform((width, height))
+        # we need to grab the tranfrom via a parent class method because
+        # we want the transform from the coordiante system the arc will be
+        # drawn in to the screen space.
+        # The transform that we get via `self.get_transform()` goes from
+        # an idealized unit-radius space to screen space
```

Comment:
```suggestion
        # We need to grab the transform via a parent class method because
        # we want the transform from the coordinate system the arc will be
        # drawn in to the screen space.
        # The transform that we get via `self.get_transform()` goes from
        # an idealized unit-radius space to screen space.
```